File: go-client-sdk/pkg/models/operations/getonecrmevent.go
package operations

import (
	"github.com/speakeasy-sdks/orb-sdks/go-client-sdk/pkg/models/shared"
)

type GetOneCrmEventQueryParams struct {
	AccessToken string `queryParam:"style=form,explode=true,name=accessToken"`
	AllFields   *bool  `queryParam:"style=form,explode=true,name=allFields"`
	ID          string `queryParam:"style=form,explode=true,name=id"`
}

type GetOneCrmEventResponseBody struct {
	Event *shared.Event `json:"event,omitempty"`
}

type GetOneCrmEventRequest struct {
	QueryParams GetOneCrmEventQueryParams
}

type GetOneCrmEventResponse struct {
	ContentType  string
	StatusCode   int64
	ResponseBody *GetOneCrmEventResponseBody
}

File: typescript-client-sdk/src/sdk/models/operations/getonecrmevent.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetOneCrmEventQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=accessToken" })
  accessToken: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=allFields" })
  allFields?: boolean;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=id" })
  id: string;
}


export class GetOneCrmEventResponseBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=event" })
  event?: shared.Event;
}


export class GetOneCrmEventRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: GetOneCrmEventQueryParams;
}


export class GetOneCrmEventResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  responseBody?: GetOneCrmEventResponseBody;
}
